cardano_node_tests/tests/tests_conway/test_pparam_update.py

```diff
@@ -245,6 +245,7 @@ def test_pparam_update(  # noqa: C901
         cluster, governance_data = cluster_lock_governance
         temp_template = common.get_test_id(cluster)
         cost_proposal_file = DATA_DIR / "cost_models_list.json"
+        proposal_records = []
```

Comment:
Something like this was also needed in PR#2444, and they decided to do it like this - https://github.com/IntersectMBO/cardano-node-tests/pull/2444/files#diff-47aea5c0f88f54444523fe50b4cbf785dceb4d5978ceea731e11872523621572R641.
Although generally I don't like `nonlocal`, in this situation it is justifiable. So please wait until we merge PR#2444.

Reply:
Merged changes from `master` and updated to include new proposal counter variable introduced by already merged PR.

Ran tests: :heavy_check_mark: 
```sh
pytest -sk "test_pparam_update" cardano_node_tests/
============================================================================================================================================================================================= test session starts =============================================================================================================================================================================================
platform linux -- Python 3.10.12, pytest-7.4.4, pluggy-1.5.0 -- /home/artur/Downloads/CardanoTests/cardano-node-tests/dev_workdir/.env/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.10.12', 'Platform': 'Linux-5.15.0-116-generic-x86_64-with-glibc2.37', 'Packages': {'pytest': '7.4.4', 'pluggy': '1.5.0'}, 'Plugins': {'order': '1.2.1', 'allure-pytest': '2.13.5', 'metadata': '3.1.1', 'xdist': '3.3.1', 'select': '0.1.2', 'hypothesis': '6.100.2', 'html': '3.2.0'}, 'cardano-node': '9.0.0', 'cardano-node rev': '2820a63dc934c6d5b5f450b6c2543b81c6476696', 'cardano-node ghc': 'ghc-8.10', 'cardano-cli': '9.0.0.0', 'cardano-cli rev': '2820a63dc934c6d5b5f450b6c2543b81c6476696', 'cardano-cli ghc': 'ghc-8.10', 'CLUSTER_ERA': 'conway', 'TX_ERA': '', 'COMMAND_ERA': 'conway', 'SCRIPTS_DIRNAME': 'conway_fast', 'ENABLE_P2P': 'False', 'MIXED_P2P': 'False', 'NUM_POOLS': '3', 'DB_BACKEND': '', 'HAS_CC': 'True', 'cardano-node-tests rev': '55dcf7d8ca07be1324bbd63a6c5680585354f38c', 'cardano-node-tests url': 'https://github.com/IntersectMBO/cardano-node-tests/tree/55dcf7d8ca07be1324bbd63a6c5680585354f38c', 'CARDANO_NODE_SOCKET_PATH': '/home/artur/Downloads/CardanoTests/cardano-node-tests/dev_workdir/state-cluster0/bft1.socket', 'cardano-cli exe': '/nix/store/gsj96vxrv2h20y97ipg1db8cnir8mghp-cardano-cli-exe-cardano-cli-9.0.0.0/bin/cardano-cli', 'cardano-node exe': '/nix/store/mbr80g2xr3c8z82afdmvwqlcqwd912mw-cardano-node-exe-cardano-node-9.0.0/bin/cardano-node', 'cardano-submit-api exe': '/nix/store/gg2n3hs6z2si1jl29i6wdfr2vjlm7q3q-cardano-submit-api-exe-cardano-submit-api-9.0.0/bin/cardano-submit-api', 'network magic': 42, 'HAS_DBSYNC': 'True', 'db-sync': '13.2.0.2', 'db-sync rev': '7dd3f60bf12c1d2853baf44e267a8dc3cb52bccf', 'db-sync ghc': 'ghc-8.10', 'db-sync exe': '/nix/store/lsb2n7s66bg6swspabigrrvn0z1x83h4-cardano-db-sync-exe-cardano-db-sync-13.2.0.2/bin/cardano-db-sync'}
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase(PosixPath('/home/artur/Downloads/CardanoTests/cardano-node-tests/.hypothesis/examples'))
rootdir: /home/artur/Downloads/CardanoTests/cardano-node-tests
configfile: pyproject.toml
plugins: order-1.2.1, allure-pytest-2.13.5, metadata-3.1.1, xdist-3.3.1, select-0.1.2, hypothesis-6.100.2, html-3.2.0
collected 1440 items / 1438 deselected / 2 selected                                                                                                                                                                                                                                                                                                                                                           

cardano_node_tests/tests/tests_conway/test_pparam_update.py::TestPParamUpdate::test_pparam_update@long 
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- live log setup ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
INFO     cardano_node_tests.tests.conftest:conftest.py:197 Changed CWD to '/home/artur/Downloads/CardanoTests/cardano-node-tests/dev_workdir/tmp/pytest-of-artur/pytest-4'.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ live log call ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
INFO     cardano_clusterlib.clusterlib_klass:clusterlib_klass.py:372 Waiting for 37.40 sec for slot no 71005.
INFO     cardano_clusterlib.clusterlib_klass:clusterlib_klass.py:372 Waiting for 61.80 sec for slot no 72025.
INFO     cardano_clusterlib.clusterlib_klass:clusterlib_klass.py:372 Waiting for 186.60 sec for slot no 73025.
PASSED
cardano_node_tests/tests/tests_conway/test_pparam_update.py::TestPParamData::test_pparam_keys PASSED
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- live log teardown ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
INFO     cardano_node_tests.utils.dbsync_conn:dbsync_conn.py:34 Closing connection to db-sync database dbsync0.


=============================================================================================================================================================================== 2 passed, 1438 deselected in 531.81s (0:08:51) ================================================================================================================================================================================

```